internal/cmd/build-uki.go

```diff
@@ -936,26 +936,21 @@ func createContainer(sourceDir, outputDir, artifactName, version string, logger
 // GetUkiCmdline returns the cmdline to be used for the kernel.
 // The cmdline can be overridden by the user using the cmdline flag.
 // For each cmdline passed, we generate a uki file with that cmdline
-// extend-cmdline will just extend the default cmdline so we only create one efi file
-// extra-cmdline will create a new efi file for each cmdline passed
+// extend-cmdline will just extend the default cmdline so we only create one efi file. Artifact name is the default one
+// extra-cmdline will create a new efi file for each cmdline passed. artifact name is generated from the cmdline
 func GetUkiCmdline(cmdlineExtend, bootBranding string, extraCmdlines []string) []utils.BootEntry {
 	defaultCmdLine := constants.UkiCmdline + " " + constants.UkiCmdlineInstall
 
-	// Extend only
+	// Override the default cmdline if the user passed one
 	if cmdlineExtend != "" {
-		cmdline := defaultCmdLine + " " + cmdlineExtend
-		return []utils.BootEntry{{
-			Cmdline:  cmdline,
-			Title:    bootBranding,
-			FileName: NameFromCmdline(constants.ArtifactBaseName, cmdline),
-		}}
+		defaultCmdLine = defaultCmdLine + " " + cmdlineExtend
 	}
 
 	// default entry
 	result := []utils.BootEntry{{
 		Cmdline:  defaultCmdLine,
 		Title:    bootBranding,
-		FileName: NameFromCmdline(constants.ArtifactBaseName, defaultCmdLine),
+		FileName: constants.ArtifactBaseName,
```

Comment:
if this is named "norole.efi" then the extraCmdlines below would be called "norole_thingshere.efi" and such, no? Will these names work when we install? What is `norole` being replaced with in that case?

Reply:
in any case, this change in this line is moot because the result is the same, in both cases due to the default cmdline, the artifact name will be norole always. I just wanted this to be more clear, but can revert it if needed.